### PR TITLE
Added suggestion to use Fluent API pattern in page object classes.

### PR DESCRIPTION
**Pull request by djangofan**
Just a suggestion to use a Fluent API in Page Object classes.   This is sometimes a helpful thing to do to make tests more human readable although it makes debugging them slightly harder.

### CODE REVIEW

**Comment by djangofan** [comment]
I'll need to amend this commit a little bit with this example to call fluent code:
driver.get( System.getProperty("testURL") + "webhp?hl=en&tab=ww" );
        GoogleSearchPage gsp = new GoogleSearchPage();
        gsp.withFluent() .....

**Comment by davehunt** [comment]
Did you want to update this pull or submit those changes in another pull?

**Comment by djangofan** [comment]
I will add that change in another separate pull.  I didn't do this pull request the usual way, by forking the project, and so I don't really know how to do additional updates on it.   Excuse me while I am learning how to properly do my commits.   I've never had trouble on any other repo until this one... not sure why... but trying to understand why...

File: 05_Best_Practices.md [hunk 1]
```diff
@@ -164,6 +164,74 @@ Python:
 Avoid Sharing State
 -------------------
 
+Consider Using a Fluent API
+-------------------
+Martin Fowler coined the term "Fluent API".  You could search the Google search page using a fluent
+API call like so:  
+
+    driver.get().withFluent().setSearchString().clickSearchButton();
```
Comment (djangofan) [inline 05_Best_Practices.md:172]: I will change this line on another commit, to the following:
      driver.get( System.getProperty("testURL") + "webhp?hl=en&tab=ww" );
      GoogleSearchPage gsp = new GoogleSearchPage();
      gsp.get().withFluent().setSearchString().clickSearchButton();

**Pull request merged**